### PR TITLE
[CU-8688ekm8c] Adjust collections to respect certain rules

### PR DESCRIPTION
**Pull request by lrojas94**

### CODE REVIEW

**Comment by JustMaier** [comment]
Task linked: [CU-8688ekm8c Adjust collection permissions](https://app.clickup.com/t/8688ekm8c)

File: src/components/Collections/AddToCollectionModal.tsx [hunk 1]
```diff
@@ -380,10 +381,16 @@ function NewCollectionForm({
...
+                  ...Object.values(CollectionMode)
+                    .map((value) =>
+                      [CollectionMode.Bookmark].some((v) => v === value)
+                        ? undefined
+                        : {
+                            value,
+                            label: getDisplayName(value),
+                          }
+                    )
+                    .filter(isDefined),
```
Comment (manuelurenah) [inline src/components/Collections/AddToCollectionModal.tsx:393]: This seems a little bit over engineered 😛 won't this yield the same result?
```suggestion
                  ...Object.values(CollectionMode)
                    .filter((v) => v !== CollectionMode.Bookmark)
                    .map((value) => ({
                      value,
                      label: getDisplayName(value),
                    })),
```

File: src/server/services/collection.service.ts [hunk 2]
```diff
@@ -603,14 +606,46 @@ export const upsertCollection = async ({
         where: { collectionId: updated.id },
         data: { publishedAt: new Date() },
       });
-    } else {
+    } else if (updated.mode !== CollectionMode.Contest) {
```
Comment (manuelurenah) [inline src/server/services/collection.service.ts:609]: I think we shouldn't allow this for `Bookmark` collections either
```suggestion
    } else if (!updated.mode) {
```

File: src/components/Collections/AddToCollectionModal.tsx (hunk 1 above)
Comment (lrojas94) [reply src/components/Collections/AddToCollectionModal.tsx:393]: Made so that if we need to filter more in the future we can 😎

**Pull request merged**